rust/gitxetcore/src/command/materialize.rs

```diff
@@ -0,0 +1,60 @@
+use clap::Args;
+use lazy::lazy_pathlist_config::{check_or_create_lazy_config, LazyPathListConfigFile};
+
+use crate::errors::{GitXetRepoError, Result};
+use crate::git_integration::git_repo::GitRepo;
+use crate::git_integration::git_wrap::{self, list_files_from_repo};
+use crate::{config::XetConfig, constants::GIT_LAZY_CHECKOUT_CONFIG};
+
+#[derive(Args, Debug)]
+pub struct MaterializeArgs {
+    #[clap(short, long)]
+    recursive: bool,
+
+    path: String,
+}
+
+pub async fn materialize_command(cfg: XetConfig, args: &MaterializeArgs) -> Result<()> {
+    // Make sure repo working directory is clean
+    let repo = GitRepo::open(cfg.clone())?;
+
+    eprintln!("Checking repo state...");
+
+    if !repo.repo_is_clean()? {
+        return Err(GitXetRepoError::InvalidOperation(
+            "Repo is dirty; commit your changes and try this operation again.".to_owned(),
+        ));
+    }
+
+    let lazy_config_path = if let Some(path) = &cfg.lazy_config {
+        path.to_owned()
+    } else {
+        let path = cfg.repo_path()?.join(GIT_LAZY_CHECKOUT_CONFIG);
+        check_or_create_lazy_config(&path).await?;
+        path
+    };
+
+    eprintln!("Materializing...");
```

Comment:
Let's move this later, like right after the path_list parsing, and add some information about the input (e.g. "Materializing {} files.").

Reply:
Also added message for no file found under the input path.